### PR TITLE
Feature/flux cluster

### PR DESCRIPTION
**Pull request by koning**
Add flux,lsf and pbs native schedulerchecks, pbs is untested. Can't install flux to pbs docker image.
Updated isort fix-style so it actually runs now.
Ran fix-style

### CODE REVIEW

**Comment by koning** [comment]
I'm getting these errors locally when trying to build a venv for linting python3.6:
```
pip requires Python '>=3.7' but the running Python is 3.6.4
You are using pip version 9.0.1, however version 23.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command. 
celery requires Python '>=3.7' but the running Python is 3.6.4
```

**Comment by lucpeterson** [comment]
I believe we decided to drop support for py3.6 in v.1.9.0 because it is past end-of-life, it was breaking our CI-testing, and dependencies were not keeping up. GitHub actions also dropped support, so this is what caused us to have to do the same.

From: Joe Koning ***@***.***>
Date: Thursday, March 2, 2023 at 12:57 PM
To: LLNL/merlin ***@***.***>
Cc: Peterson, Luc ***@***.***>, Review requested ***@***.***>
Subject: Re: [LLNL/merlin] Feature/flux cluster (PR #402)

I'm getting these errors locally when trying to build a venv for linting python3.6:

pip requires Python '>=3.7' but the running Python is 3.6.4

You are using pip version 9.0.1, however version 23.0.1 is available.

You should consider upgrading via the 'pip install --upgrade pip' command.

celery requires Python '>=3.7' but the running Python is 3.6.4

**Comment by koning** [comment]
OK, the lint check is failing and it has a py36 target. All of the lint checks pass locally. Can the black version be changes to match the new one?
 
```
Run python3 -m black --check --line-length $MAX_LINE_LENGTH --target-version py36 merlin
would reformat /home/runner/work/merlin/merlin/merlin/ascii_art.py
would reformat /home/runner/work/merlin/merlin/merlin/common/tasks.py
would reformat /home/runner/work/merlin/merlin/merlin/study/dag.py

Oh no! 💥 💔 💥
3 files would be reformatted, 83 files would be left unchanged.
Error: Process completed with exit code 1.
```

**Comment by koning** [comment]
Lint test is working now.

**Comment by lucpeterson** [comment]
Huzzah! I’m good with this

File: merlin/examples/workflows/flux/flux_par_native_test.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+description:
+  description: A simple ensemble of parallel MPI jobs run by flux.
+  name: flux_par
+
+batch:
+  type: flux
+  nodes: 1
+  flux_exec: flux exec -r "0-1"
+  flux_start_opts: -o,-S,log-filename=flux_par.out
+  nodes: 3
```
Comment (lucpeterson) [inline merlin/examples/workflows/flux/flux_par_native_test.yaml:10]: duplicate of line 7

File: merlin/study/batch.py [hunk 2]
```diff
@@ -160,29 +255,37 @@ def batch_worker_launch(
...
     if launch_pre:
         launch_command = f"{launch_pre} {launch_command}"
 
+    LOG.debug(f"launch_command = {launch_command}")
+
     worker_cmd: str = ""
     if btype == "flux":
         flux_path: str = get_yaml_var(batch, "flux_path", "")
+        if "/" in flux_path:
+            flux_path += "/"
```
Comment (lucpeterson) [inline merlin/study/batch.py:272]: what's this for?

File: tests/integration/test_definitions.py [hunk 3]
```diff
@@ -23,6 +23,8 @@ def define_tests():
     of (shell command, condition(s) to satisfy).
     """
     celery_regex = r"(srun\s+.*)?celery\s+(-A|--app)\s+merlin\s+worker\s+.*"
```
Comment (lucpeterson) [inline tests/integration/test_definitions.py:25]: maybe change this to celery_slurm_regex for consistency?

File: merlin/study/batch.py (hunk 2 above)
Comment (lucpeterson) [reply merlin/study/batch.py:272]: oh I see this was moved from earlier

File: merlin/examples/workflows/flux/flux_par_native_test.yaml (hunk 1 above)
Comment (koning) [reply merlin/examples/workflows/flux/flux_par_native_test.yaml]: Fixed

File: tests/integration/test_definitions.py (hunk 3 above)
Comment (koning) [reply tests/integration/test_definitions.py]: OK, I changed that name.

**Review by lucpeterson — APPROVED** [review]
LGTM

**Pull request merged**